pkg/coscheduling/core/core.go

```diff
@@ -112,7 +119,46 @@ func (pgMgr *PodGroupManager) PreFilter(ctx context.Context, pod *corev1.Pod) er
 		return fmt.Errorf("cannot found engough pods, "+
 			"current pods number: %v, minMember of group: %v", len(pods), pg.Spec.MinMember)
 	}
-	return nil
+
+	// TODO(cwdsuzhou): This resource check can not absolutely to avoid unschedulable group.
+	// Only PreFilter resource constraints so even if a PodGroup passed here,
+	// it may not necessarily pass Filter due to other constraints such as affinity/taints.
+	nodes, err := pgMgr.snapshotSharedLister.NodeInfos().List()
+	if err != nil {
+		return err
+	}
+	if !pgMgr.enableResourcePreCheck {
+		return nil
+	}
```

Comment:
This should be applied only when MinResource is nil (L136).

I'd suggest to refine the logic flow like this:

```go
if _, ok := pgMgr.permittedPG.Get(pgFullName); ok {
    return nil
}

if pg.Spec.MinResources != nil {
    // logic here, and add it to permitted cache if needed
    return Success or Failure
}

if !pgMgr.enableResourcePreCheck {
    return nil
}

// Logic to handle PreCheck on Pod's req * minNum
```